src/components/BulkOperations/index.jsx

```diff
@@ -0,0 +1,208 @@
+      Header: fieldName, accessor: fieldName, width: 400, size: 400,
+    }));
+
+    return (
+      <div data-testid="file-preview" className="data-container-excel">
+        <div className="row justify-content-between border-bottom">
+          <div className="col-auto">
+            <span className="font-weight-bold">{file.name}</span><br />
+            {parsed.data.length} rows - {file.size} B
+          </div>
+          <div className="col-auto">
+            <button data-testid="process-file" className="btn btn-outline-primary mr-2" onClick={handleSubmit}>
+              Process File
+            </button>
+            <button data-testid="upload-new" className="btn btn-outline-primary" onClick={handleUploadNew}>
+              Upload New File
+            </button>
+          </div>
+        </div>
+
+        <div className="mt-3">
+          <DataTable
+            isPaginated
+            initialState={{
+              pageSize: 2,
+            }}
+            itemCount={parsed.data.length}
+            data={parsed.data}
+            columns={columns}
+          >
+            <DataTable.Table />
+            <DataTable.EmptyTable content="No results found" />
+            <DataTable.TableFooter />
+          </DataTable>
+        </div>
+      </div>
+    );
+  }
+
+  useEffect(() => {
+    fetchBulkOpTasks().then(() => setIsLoading(false));
+  }, []);
+
+  if (isLoading) {
+    return (
+      <div className="bulk-operations-spinner">
+        <LoadingSpinner />
+      </div>
+    );
+  }
+
+  return (
+    <div className="mx-5 my-5">
+      {getAlert()}
+      <SelectMenu className="mb-3" defaultMessage="Choose a Bulk Operation">
+        {
+                    Object.entries(availableOperations).map(([slug, title]) => (
+                      <MenuItem actionid={slug} defaultSelected={slug === bulkOperationId} onClick={e => setBulkOperationId(e.currentTarget.getAttribute('actionid'))}>
+                        {title}
+                      </MenuItem>
+                    ))
+                }
+      </SelectMenu>
+
+      <Collapsible title={getCollapsibleTitle()}>
+        {filteredHistoricalRecords().map(rec => (
+          <>
+            <div className="row justify-content-between">
+              <div className="col-auto">
+                <a href={rec.csv_file}>{rec.csv_file.split('/').pop()}</a><br />
+                <span className="small">{formatDate(rec.created)}</span>
+              </div>
+              <div className="col-auto d-flex flex-column align-items-end">
+                <span className={classNames(
+                  {
+                    badge: true,
+                    'badge-danger': rec.status == 'failed',
+                    'badge-success': rec.status == 'completed',
+                    'badge-warning': rec.status == 'processing',
+                    'badge-info': rec.status == 'pending',
+                  },
+                )}
+                >
+                  {rec.status}
```

Comment:
nit: Please convert the status into sentence case, all small-caps is not consistent with how the statuses are displayed on other places (like Course Statuses).

Reply:
+1.  Also, can we use a dictionary mapping? I personally prefer that approach as it looks cleaner.

Reply:
> Capitalized status.

Where?

Reply:
[Here](https://github.com/openedx/frontend-app-publisher/pull/1011/files#diff-4f7b37abdf9a555b283112cbeab8c3b9538a62ee13e8aa7dedef87f5691c034cR194)